### ISSUE TITLE
Exception in thread "Thread-4" com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception

### ISSUE THREAD

**Issue by SebRut** [googleapis/java-dialogflow#27]
I am trying to access a list of intents for a dialogflow project using a service json file. Running my program on my development MacBooks works flawlessly but using the same .jar on the target machines running Windows fails with the exception detailed below.
#### Environment details

- OS: Windows 7 64bit
- Java version:
`java version "1.8.0_131"`
`Java(TM) SE Runtime Environment (build 1.8.0_131-b11)`
`Java HotSpot(TM) 64-Bit Server VM (build 25.131-b11, mixed mode)`
- google-cloud-java version(s):
`com.google.cloud:google-cloud-dialogflow:0.70.0-alpha`

#### Stacktrace

```
Exception in thread "Thread-4" com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception
at com.google.api.gax.rpc.ApiExceptionFactory.createException(ApiExceptionFactory.java:69)
at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:72)
at com.google.api.gax.grpc.GrpcApiExceptionFactory.create(GrpcApiExceptionFactory.java:60)
at com.google.api.gax.grpc.GrpcExceptionCallable$ExceptionTransformingFuture.onFailure(GrpcExceptionCallable.java:97)
at com.google.api.core.ApiFutures$1.onFailure(ApiFutures.java:68)
at com.google.common.util.concurrent.Futures$4.run(Futures.java:1123)
at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:435)
at com.google.common.util.concurrent.AbstractFuture.executeListener(AbstractFuture.java:900)
at com.google.common.util.concurrent.AbstractFuture.addListener(AbstractFuture.java:634)
at com.google.common.util.concurrent.ForwardingListenableFuture.addListener(ForwardingListenableFuture.java:45)
at com.google.api.core.ApiFutureToListenableFuture.addListener(ApiFutureToListenableFuture.java:52)
at com.google.common.util.concurrent.Futures.addCallback(Futures.java:1135)
at com.google.api.core.ApiFutures.addCallback(ApiFutures.java:63)
at com.google.api.gax.grpc.GrpcExceptionCallable.futureCall(GrpcExceptionCallable.java:67)
at com.google.api.gax.rpc.AttemptCallable.call(AttemptCallable.java:81)
at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:111)
at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:58)
at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:75)
```

#### Code snippet
_Setup code_
```java
ServiceAccountCredentials credentials = ServiceAccountCredentials.fromStream(new FileInputStream(serviceJsonPath));
IntentsSettings settings = IntentsSettings.newBuilder().setCredentialsProvider(FixedCredentialsProvider.create(credentials)).build();
```

_The program fails somewhere here_
```java
IntentsClient client = IntentsClient.create(settings);
Iterable<Intent> intents = client.listIntents(String.format("projects/%s/agent", projectId)).iterateAll();
```

#### External references such as API reference guides used

- https://dialogflow.com/docs/reference/api-v2/rest/v2/projects.agent.intents/list

#### Any additional information below

The Windows machine needs a proxy to connect to the internet which I configured using `http.proxyHost`, `http.proxyPort`, `https.proxyHost` and `https.proxyPort`.

**Comment by athakor** [comment]
> The Windows machine needs a proxy to connect to the internet which I configured using http.proxyHost, http.proxyPort, https.proxyHost and https.proxyPort.

@SebRut I guess the cause`(com.google.api.gax.rpc.UnavailableException: io.grpc.StatusRuntimeException: UNAVAILABLE: io exception)` of issue is your proxy, might be your request is not properly passing through the proxy. I tried to get the list of intents for a `Dialogflow `with `windows `+ `proxy `and it's working fine please try the following sample to get the list of intents.

**Sample :**
```javascript
import com.google.api.core.ApiFunction;
import com.google.api.gax.core.FixedCredentialsProvider;
import com.google.api.gax.rpc.TransportChannelProvider;
import com.google.auth.oauth2.ServiceAccountCredentials;
import com.google.cloud.dialogflow.v2.Intent;
import com.google.cloud.dialogflow.v2.IntentsClient;
import com.google.cloud.dialogflow.v2.IntentsSettings;
import com.google.cloud.dialogflow.v2.stub.SessionsStubSettings;
import io.grpc.HttpConnectProxiedSocketAddress;
import io.grpc.ManagedChannelBuilder;
import io.grpc.ProxiedSocketAddress;
import io.grpc.ProxyDetector;
import java.io.FileInputStream;
import java.io.IOException;
import java.net.InetSocketAddress;
import java.net.SocketAddress;
import javax.annotation.Nullable;

public class DialogflowIntentsClientExample {

  private static final String PROXY_HOST = "your-proxy-host";
  private static final Integer PROXY_PORT = 8080; //your-proxy-port
  private static final String PROXY_USERNAME = "your-proxy-username";
  private static final String PROXY_PASSWORD = "your-proxy-password";
  private static final String PROJECT_ID = "your-project-id";
  private static final String JSON_KEY_PATH = "/path/to/your/key.json"";

  public static void main(String[] args) throws IOException {
    Iterable<Intent> intents =
        getService().listIntents(String.format("projects/%s/agent", PROJECT_ID)).iterateAll();
    for (Intent intent : intents) {
      System.out.println(intent);
    }
  }

  public static IntentsClient getService() throws IOException {
    TransportChannelProvider transportChannelProvider =
        SessionsStubSettings.defaultGrpcTransportProviderBuilder()
            .setChannelConfigurator(
                new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
                  @Override
                  public ManagedChannelBuilder apply(ManagedChannelBuilder managedChannelBuilder) {
                    return managedChannelBuilder.proxyDetector(
                        new ProxyDetector() {
                          @Nullable
                          @Override
                          public ProxiedSocketAddress proxyFor(SocketAddress socketAddress) {
                            return HttpConnectProxiedSocketAddress.newBuilder()
                                .setUsername(PROXY_USERNAME)
                                .setPassword(PROXY_PASSWORD)
                                .setProxyAddress(new InetSocketAddress(PROXY_HOST, PROXY_PORT))
                                .setTargetAddress((InetSocketAddress) socketAddress)
                                .build();
                          }
                        });
                  }
                })
            .build();
    IntentsSettings intentsSettings =
        IntentsSettings.newBuilder()
            .setTransportChannelProvider(transportChannelProvider)
            .setCredentialsProvider(
                FixedCredentialsProvider.create(
                    ServiceAccountCredentials.fromStream(new FileInputStream(JSON_KEY_PATH))))
            .build();
    return IntentsClient.create(intentsSettings);
  }
}
```


**Output :**
```
name: "projects/cloud-test-xxxxx-xxxxx/agent/intents/87d25bc3-c633-406f-8b98-d3b0e249c786"
display_name: "Default Welcome Intent"
priority: 500000
events: "WELCOME"
action: "input.welcome"
messages {
  text {
    text: "Hi! How are you doing?"
    text: "Hello! How can I help you?"
    text: "Good day! What can I do for you today?"
    text: "Greetings! How can I assist?"
  }
}

name: "projects/cloud-test-xxxxx-xxxxx/agent/intents/8eb85780-5218-4626-976f-6557c8998aca"
display_name: "test_intent"
priority: 500000
webhook_state: WEBHOOK_STATE_ENABLED
action: "action"
messages {
  platform: ACTIONS_ON_GOOGLE
  basic_card {
    title: "test"
    subtitle: "test"
    formatted_text: "test"
    image {
      image_uri: "http://test.jpg"
      accessibility_text: "test"
    }
    buttons {
      title: "test"
      open_uri_action {
        uri: "http://test"
      }
    }
  }
}
default_response_platforms: ACTIONS_ON_GOOGLE

name: "projects/cloud-test-xxxxx-xxxxx/agent/intents/cee713c7-6fd0-49e7-986c-5958f8b4eaf8"
display_name: "Default Fallback Intent"
priority: 500000
is_fallback: true
action: "input.unknown"
messages {
  text {
    text: "I didn\'t get that. Can you say it again?"
    text: "I missed what you said. What was that?"
    text: "Sorry, could you say that again?"
    text: "Sorry, can you say that again?"
    text: "Can you say that again?"
    text: "Sorry, I didn\'t get that. Can you rephrase?"
    text: "Sorry, what was that?"
    text: "One more time?"
    text: "What was that?"
    text: "Say that one more time?"
    text: "I didn\'t get that. Can you repeat?"
    text: "I missed that, say that again?"
  }
}
```

**Comment by SebRut** [comment]
I'm not able to test this anymore.
